packages/dd-trace/src/appsec/reporter.js

```diff
@@ -144,6 +145,30 @@ function reportAttack (attackData) {
   rootSpan.addTags(newTags)
 }
 
+function reportSchemas (derivatives) {
+  if (!derivatives) {
+    return
+  }
+
+  const store = storage.getStore()
+  const req = store?.req
+  const rootSpan = web.root(req)
+
+  if (!rootSpan) return
+
+  const tags = {}
+  derivatives.forEach((address) => {
+    for (const [key, value] of Object.entries(address)) {
+      if (key.includes('_dd.appsec.s.req')) {
+        const gzippedValue = zlib.gzipSync(JSON.stringify(value))
```

Comment:
Is there a way for us to not do this synchronously ? zlib will use a different thread and here the whole process will wait for that operation :/